src/oss/confluent.sh

```diff
@@ -978,6 +979,107 @@ print_current() {
+      cd $repo && git fetch --tags ; git checkout $cp_version 2> /dev/null
+    fi
+}
+
+
+is_demo_name_valid() {
+    local demo_name="${1}"
+    local repo="${2}"
+    local subcommand="${3}"
+
+    if [ -z $demo_name ]; then
+      echo "'confluent demo $subcommand' requires an argument that specifies the demo name. Please run 'confluent demo list' to see available demos"
+      return 1
+    elif [ ! -d $demo_name ]; then
+      echo "Demo subfolder $demo_name does not exist in ${confluent_home}/$repo. Please run 'confluent demo list' to see available demos"
+      return 1
+    else
+      return 0
+    fi
+
+}
+
+demo_command() {
+    local subcommand="${1}"
+    local demo_name="${2}"
+    local repo="quickstart-demos"
+
+    if [[ $subcommand == "" || $subcommand == "help" ]]; then
+      demo_usage
+      return
+    fi
+
+    # Check network connectivity
+    local network_status=0
+    if ping -q -c 1 -W 1 github.com >/dev/null; then
+      network_status=1
+    else
+      network_status=0
+    fi
+  
+    # Clone the repo
+    if [ ! -d ${confluent_home}/$repo ]; then
+      if [ $network_status == 1 ]; then
+        git clone https://github.com/confluentinc/$repo.git ${confluent_home}/$repo
+        status=$?
+        if [[ ${status} -ne 0 ]]; then
+            echo "'git clone https://github.com/confluentinc/$repo.git ${confluent_home}/$repo' failed. Please check your git access"
+            return
+        fi
+        cp_version=$(confluent version | awk -F':' '{print $2;}' | awk '$1 > 0 { print substr($1,1,3)}')
+        cd ${confluent_home}/$repo && git checkout $cp_version 2> /dev/null
+      else
+        echo "'confluent demo $subcommand $demo_name' requires network connectivity. Please try again when you are connected."
+        return
+      fi
+    fi
+
+    cd ${confluent_home}/$repo
+    if [[ ${status} -ne 0 ]]; then
+        echo "Cannot find ${confluent_home}/$repo.  Exiting"
+        return
+    fi
+
+    cp_version=$(confluent version | awk -F':' '{print $2;}' | awk '$1 > 0 { print substr($1,1,3)}')
+    if [[ $subcommand == "list" ]]; then
+      check_demo_repo_uptodate $network_status "${confluent_home}/$repo"
+      echo -e "Available demos from ${confluent_home}/$repo (start a demo 'confluent demo start <demo-name>':"
+      ls | grep -v "README.md" | grep -v "utils" | grep -v "LICENSE"
+    elif [[ $subcommand == "refresh" ]]; then
+      if [ $network_status == 1 ]; then
+        cd ${confluent_home}/$repo && git fetch --tags ; git checkout $cp_version
+      else
+        echo "Running 'confluent demo $subcommand $demo_name' requires network connectivity. Please try again when you are connected."
+        return
+      fi
+    elif [[ $subcommand == "start" ]]; then
+      check_demo_repo_uptodate $network_status "${confluent_home}/$repo"
+      is_demo_name_valid "$demo_name" "$repo" "$subcommand" || die
+      cd $demo_name
```

Comment:
Using relative paths here and within `start.sh` and `stop.sh` complicates things. 

However, seems like all the required paths, including the demo's home as well as `$CONFLUENT_HOME` can be extracted during runtime. I'd think this is more robust.

Reply:
@kkonstantine , can you please clarify what is the recommended change?  Right now it does appear to be extracted during runtime based on value of `${confluent_home}`